### ISSUE TITLE
Checkbox and Checkbox group read only inconsistency

### ISSUE THREAD

**Issue by timbo86** [vaadin/flow-components#3522]
### Description

The readonly style and behaviour of the checkbox and checkbox group differ to the ones from the radio button and radio button group.

### Expected outcome

Checkbox: Read only property should be reflected to attribute like in radio button.
Checkbox group: Read only sets items disabled instead of settings them to read only as well, like in radio button group.

### Minimal reproducible example

Create simple view:

```
@Route("checkbox")
public class CheckboxView extends Div {

    public CheckboxView() {
        Checkbox checkbox = new Checkbox("Read only checkbox");
        checkbox.setReadOnly(true);
        add(checkbox);

        CheckboxGroup<String> checkboxGroup = new CheckboxGroup<>("Checkbox group");
        checkboxGroup.setItems("First", "Second");
        checkboxGroup.setReadOnly(true);
        add(checkboxGroup);

        RadioButtonGroup<String> radioButtonGroup = new RadioButtonGroup<>("Radio button group");
        radioButtonGroup.setItems("First", "Second");
        radioButtonGroup.setReadOnly(true);
        add(radioButtonGroup);
    }

}
```

### Steps to reproduce

Inspect single checkbox in browsers dev tools: Attribute is missing, so css selectors on read only will fail.
Compare both groups in dev tools: Checkbox group has disabled attribute, but radio button group has readonly attribute.

### Environment

Vaadin version(s): 23.1.2

### Browsers

Chrome

**Comment by web-padawan** [comment]
> Checkbox: Read only property should be reflected to attribute like in radio button.

This is not a bug. Currently, `vaadin-checkbox` does not support `readonly`: https://github.com/vaadin/web-components/issues/688

> Checkbox group: Read only sets items disabled instead of settings them to read only as well, like in radio button group

This is a bug. There was a similar issue fixed for `RadioButtonGroup` in #2526

**Comment by web-padawan** [comment]
Update: it turns out we don't have `readonly` in `vaadin-checkbox-group` either, so it's technically a missing feature.

One notable difference compared to `vaadin-radio-group`: the radio-button can't be unchecked, so for `vaadin-radio-group` the `readonly` state is set by disabling all the individual radio buttons, except the one that is checked.

However, `vaadin-checkbox` can be unchecked. So the current behavior of `setReadOnly()` kinda makes sense.

Closing this in favor of https://github.com/vaadin/web-components/issues/688 as it's essentially a duplicate.
